265 adding ipt required fields to metadata form

sjbruce opened the pull request:
Added a biological tab for Geographic Description, Sampling Description, Study Extent and Method Steps (repeating section).

Updated EML generation to include new elements.

n-a-t-e commented:
This looks good! 

It would be good to give this "Add Item" button some context, like if items are method steps, maybe this can be in a section called "Methods" and items can be "Steps" or something like that.

![Screen Shot 2023-11-30 at 4 48 05 PM](https://github.com/cioos-siooc/metadata-entry-form/assets/26209011/04611606-0c74-435c-b83c-1c2432db636c)

sjbruce commented:
Thanks, only getting back to this now.

As suggested, I've renamed the buttons to give more context and moved the method steps themselves into a section with descriptive text.  I've also added key to the steps so they can be enumerated on the page.

I guess this could be reworked to behave like contacts or instruments - for the moment I've modeled it after the resources tab.

n-a-t-e commented:
Looks good! Tiny thing - looks like "Move up" is correctly greyed out when theres a single method step, but "Move down" is still clickable (doesn't do anything)

<img width="514" alt="Screen Shot 2023-12-14 at 3 07 51 PM" src="https://github.com/cioos-siooc/metadata-entry-form/assets/26209011/2e057d29-ed48-4b34-b94a-b3d89185507c">

n-a-t-e commented on src/utils/emlTemplate.j2:
```diff
@@ -89,40 +92,65 @@
+            <methodStep>
+                <description>
+                    <section>
+                        <title>{{ method.title.en }}</title>
```
My text editor (VSCode) is complaining about <title> being here. do you have any links to info on this <title> tag?

n-a-t-e replied:
also possible my editor is using the wrong schema files, if this works in the IPT thats good enough

sjbruce replied:
I consulted a few different sources but ultimately based my structure on the following example: https://ediorg.github.io/data-package-best-practices/methods.html

Based on the idea that the description for a methodStep element is a section and a section has a title.

This seems to bear out in the EML schema detailed here, but I'm no expert in this either - the following validator doesn't complain about it: https://knb.ecoinformatics.org/emlparser/

It does have some opinions on roles being specified for associated party: 
`The content of element 'associatedParty' is not complete. One of '{userId, role}' is expected.`

fostermh commented on src/components/FormComponents/MethodSteps.jsx:
At first glance, there appears to be quite a bit of overlap between 'Method' from EML and 'Lineage' from iso1911.  Specifically the processing step.
- https://wiki.esipfed.org/LI_Lineage
- https://wiki.esipfed.org/LI_ProcessStep

fostermh commented on src/utils/emlTemplate.j2:
```diff
@@ -89,40 +92,65 @@
+                </description>
+            </methodStep>
+            {% endfor %}
+            <sampling>
```
The sampling section seems to overlap with the following iso19115 concepts:
- https://wiki.esipfed.org/EX_Extent
- https://wiki.esipfed.org/EX_TemporalExtent
- https://wiki.esipfed.org/EX_GeographicDescription
- https://wiki.esipfed.org/MD_MaintenanceInformation

fostermh commented on src/utils/emlTemplate.j2:
```diff
@@ -89,40 +92,65 @@
+                        <para>{{ record.biological.studyExtent.en }}</para>
+                    </description>
+                </studyExtent>
+                <samplingDescription>
```
samplingDescription also seems to overlap with the iso lineage section

fostermh commented on src/utils/emlTemplate.j2:
```diff
@@ -70,6 +72,7 @@
         </distribution>
         <coverage>
             <geographicCoverage>
+                <geographicDescription>{{ record.biological.geographicDescription.en }}</geographicDescription>
```
overlap with iso https://wiki.esipfed.org/EX_GeographicDescription

fostermh reviewed (commented):
highlighting a few areas I think biological data and iso19115 schema overlap

fostermh commented:
replaced by https://github.com/cioos-siooc/metadata-entry-form/pull/322
